src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -950,42 +975,66 @@ struct consume_to_store {
 
         switch (*key_type) {
         case topic_key_type::noop:
-            co_return;
+            break;
         case topic_key_type::schema: {
             std::optional<schema_value> val;
             if (!record.value().empty()) {
                 val.emplace(from_json_iobuf<schema_value_handler<>>(
                   record.release_value()));
             }
-            co_return co_await apply(
+            co_await apply(
+              offset,
               from_json_iobuf<schema_key_handler<>>(std::move(key)),
               std::move(val));
+            break;
         }
         case topic_key_type::config: {
             std::optional<config_value> val;
             if (!record.value().empty()) {
                 val.emplace(from_json_iobuf<config_value_handler<>>(
                   record.release_value()));
             }
-            co_return co_await apply(
+            co_await apply(
               from_json_iobuf<config_key_handler<>>(std::move(key)), val);
+            break;
         }
         case topic_key_type::delete_subject:
-            co_return co_await apply(
+            co_await apply(
               from_json_iobuf<delete_subject_key_handler<>>(std::move(key)),
               from_json_iobuf<delete_subject_value_handler<>>(
                 record.release_value()));
+            break;
         }
+
+        co_await _store.replay(offset);
     }
 
-    ss::future<> apply(schema_key key, std::optional<schema_value> val) {
+    ss::future<> apply(
+      model::offset offset, schema_key key, std::optional<schema_value> val) {
         if (key.magic != 0 && key.magic != 1) {
             throw exception(
               error_code::topic_parse_error,
               fmt::format("Unexpected magic: {}", key));
         }
+
+        // Out-of-place events happen when two writers collide.  First writer
+        // wins: disregard subsequent events whose seq field doesn't match
+        // their actually offset.
+        if (offset != key.seq) {
+            vlog(
+              plog.info,
+              "Ignoring out of order schema_key {} (at offset {})",
```

Comment:
Nit: I think "SCHEMA_KEY" is already part of the textified key, so some of the logging might be redundant.

This should be handled gracefully, so perhaps debug is a better log level?

Reply:
Tweaked to debug level + removed redundant part.